integration_tests/test_search.py

```diff
@@ -59,6 +63,19 @@
 #     self.book_by_someone_else,      # match across fields (no 'biography')
 # ]
 
+def known_to_fail(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        try:
+            ignore = f(*args, **kwargs)
+        except Exception, e:
+            SearchTest.expected_failures.append(f)
```

Comment:
Appending the test function seems a bit weird, but I guess you can get its name?

Reply:
Yes, it's used at the end to show the names and line numbers of the tests that need attention -- the sort of service provided by nose when a test actually fails.